### ISSUE TITLE
OPENAI_API_BASE env won't work

### ISSUE THREAD

**Issue by Steven-Luo** [truera/trulens#628]
OPENAI_API_BASE env won't work, both in environment, or below

```python
from trulens_eval.feedback.provider.openai import OpenAI as fOpenAI

fopenai = fOpenAI(
    api_base='https://xxxx/openai/v1'
)
```

when I use AzureOpenAI, deployment_id is required when create AzureOpenAI object. But with deployment_id filled, another error thrown when recording query

```python
from trulens_eval.feedback.provider.openai import AzureOpenAI

import os

os.environ['OPENAI_API_KEY'] = 'xxx'
os.environ['OPENAI_API_BASE'] = 'https://xx/v1'

fopenai = AzureOpenAI(deployment_id='xxx')
...

with tru_rag as recording:
    for query in tqdm(eval_questions):
        rag.query(query['question'])
```
`openai request failed <class 'TypeError'>=Completions.create() got an unexpected keyword argument 'deployment_id'. Retries remaining=3.` thrown when executing `rag.query`

package version:

trulens-eval==0.18.2
openai==1.3.7

**Comment by Steven-Luo** [comment]
Finally I got a workaround:

```python
from openai import OpenAI
import os

os.environ['OPENAI_API_KEY'] = 'xxx'

client = OpenAI(base_url='https://xxx/v1')
openai_client = OpenAIClient(client=client)
openai_endpoint = OpenAIEndpoint(client=openai_client)
fopenai = fOpenAI(endpoint=openai_endpoint)
```

**Comment by daniel-huang-1230** [comment]
Hi @Steven-Luo,  

Thanks for raising the issue and glad to hear you have gotten a workaround!

So the issue you were encountering in `AzureOpenAI` was due to the breaking changes introduced in the new v1.x openai api. 
Specifically, the removal of `openai.Deployment (Previously used for Azure OpenAI)
`

See for migration examples: https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/migration?tabs=python%2Cdalle-fix#use-your-data. 

We'll be adding API-parity classes in the upcoming week as well. I hope this is helpful.

**Comment by Ethan-Otto** [comment]
@Steven-Luo I tried your fix but I couldn't get it to work. The url is just the Azure end point right?

**Comment by joshreini1** [comment]
@Steven-Luo @Ethan-Otto The fix for this is now released in trulens_eval==0.18.3. Please upgrade and let us know if you have any persistent issues.

Thanks!
